sentry_sdk/profiler.py

```diff
@@ -305,13 +304,22 @@ def __init__(
         self.scheduler = scheduler
         self.transaction = transaction
         self.hub = hub
+        self.active_thread_id = None  # type: Optional[int]
         self._start_ns = None  # type: Optional[int]
         self._stop_ns = None  # type: Optional[int]
 
         transaction._profile = self
 
     def __enter__(self):
         # type: () -> None
+        hub = self.hub or sentry_sdk.Hub.current
+
+        _, scope = hub._stack[-1]
+        old_profile = scope.profile
+        scope.profile = self
+
+        self._context_manager_state = (hub, scope, old_profile)
```

Comment:
maybe cleaner to do `configure_scope/scope.profile = profile` in the `start_profiling` ctx manager below? Then you don't need this extra state.

Reply:
That context manager is going to be replaced in an upcoming change to enable profiling for all transactions since I need to hook into the transaction which does not allow me to use the context manager.